shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -241,21 +241,39 @@ public boolean sendKeyEvent(KeyEvent event) {
           // TODO(garyq): Explore how to obtain per-character direction. The
           // isRTLCharAt() call below is returning blanket direction assumption
           // based on the first character in the line.
-          boolean isRtl = mLayout.isRtlCharAt(mLayout.getLineForOffset(selStart));
-          try {
-            if (isRtl) {
-              Selection.extendRight(mEditable, mLayout);
-            } else {
-              Selection.extendLeft(mEditable, mLayout);
+
+          Boolean selectedCharIsEmoji = false;
+          if (selStart > 1) {
+            final String emo_regex =
+                "([\\u20a0-\\u32ff\\ud83c\\udc00-\\ud83d\\udeff\\udbb9\\udce5-\\udbb9\\udcee])";
```

Comment:
Where did this regex come from?  Are you sure this will match exactly the characters that you want it to match?

Reply:
Thanks for your review!
I'm now doing research on the regex that matches all needed characters.